src/main/java/com/android/volley/Request.java

```diff
@@ -619,10 +648,12 @@ public void deliverError(VolleyError error) {
      * @param response received from the network
      */
     /* package */ void notifyListenerResponseReceived(Response<?> response) {
+        NetworkRequestCompleteListener listener;
         synchronized (mLock) {
-            if (mRequestCompleteListener != null) {
-               mRequestCompleteListener.onResponseReceived(this, response);
-            }
+            listener = mRequestCompleteListener;
+        }
+        if (listener != null) {
+            listener.onResponseReceived(this, response);
```

Comment:
I should have thought of this earlier, but since the listener doesn't need the request, just the CacheKey, should we change this (and the NoUsableResponse below) to send only that, i.e. onResponseReceived(getCacheKey, response)?

Reply:
That seems reasonable, but I think it's out of the scope of this pull request. (And not something we urgently need to resolve before the next public release, since this API for network responses is package-private, and since I don't think it's harmful to pass the request on even if it's unused).